nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/TestUpload.java

```diff
@@ -155,6 +155,27 @@ public void itCanAttachABatchToADoc() {
     }
 
     @Test
+    public void itCantAttachAnExistingBatchToADoc() {
+        //Upload file
+        BatchUpload batchUpload = nuxeoClient.fetchUploadManager();
+        String batchID = batchUpload.getBatchId();
+        assertNotNull(batchID);
+        File file = FileUtils.getResourceFileFromContext("sample.jpg");
+        batchUpload = batchUpload.upload(file.getName(), file.length(), "jpg", batchID, "1", file);
+        assertNotNull(batchUpload);
+
+        // Getting a doc and attaching the existing batch file
+        Document doc = new Document("file", "File");
+        doc.setPropertyValue("dc:title", "new title");
+        doc = nuxeoClient.repository().createDocumentByPath("/folder_1", doc);
+        assertNotNull(doc);
+        doc.setPropertyValue("file:content", batchUpload.getBatchBlob(batchID));
```

Comment:
Be aware that `fetchUploadManager` create a new batch upload by making a request to server. You may have error due to file index which could be different.
You can directly instantiate your `BatchBlob` by giving it `batchId` and `fileId`.